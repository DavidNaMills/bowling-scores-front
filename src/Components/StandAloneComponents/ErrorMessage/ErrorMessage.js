import React from 'react';
import classes from './ErrorMessage.module.scss';

const ErrorMessage = ({ msg, close = null }) => (
    <div className={classes.errorMessage}>
        {close &&
            <div className={classes.errorMessage__closeBar}>
                <p onClick={close} className={classes.errorMessage__closeBar_cls}>CLOSE</p>
            </div>
        }
        <p className={classes.errorMessage__msg}>{msg}</p>
    </div>
);

export default ErrorMessage;