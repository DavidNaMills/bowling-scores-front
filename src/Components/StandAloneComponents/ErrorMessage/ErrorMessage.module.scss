@import '../../../styles/colors.scss';

.errorMessage{
    border: solid $btn_lightRed_bkg_active_hover 3px;
    background: $btn_lightRed_bkg_active;


    &__msg{
        text-align: center;
        color: $btn_font_clr;
        font-size: 20px;
        font-weight: bold;
    }

    &__closeBar{
        padding:0;
        display: flex;
        justify-content: left;
        
        &_cls{
            font-size: 12px;
            color: $btn_danger_bkg_active_hover;
        }
    }
}