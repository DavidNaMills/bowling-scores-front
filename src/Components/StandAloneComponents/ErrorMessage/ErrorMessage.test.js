import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ErrorMessage from './ErrorMessage';

describe('<ErrorMessage /> test suite', () => {
    //renders a message
    const testMsg = 'I am a long test message';
    const mockClose = jest.fn();

    it('renders a message', () => {
        const wrapper = shallow(<ErrorMessage msg={testMsg} />)
        expect(wrapper.find('.errorMessage').length).toBe(1);
        expect(wrapper.find('.errorMessage__msg').length).toBe(1);
        expect(wrapper.find('p').text()).toEqual(testMsg);
    });

    it('shows the close button', () => {
        const wrapper = shallow(<ErrorMessage
            msg={testMsg}
            close={mockClose}
        />);
        expect(wrapper.find('.errorMessage__closeBar').length).toBe(1);
        expect(wrapper.find('p').length).toBe(2);
        expect(wrapper.find('p').at(0).text()).toEqual('CLOSE');
    });
    
    
    it('shows the close button', () => {
        const wrapper = shallow(<ErrorMessage
            msg={testMsg}
            close={mockClose}
        />);
        wrapper.find('p').at(0).simulate('click');
        expect(wrapper.find('p').at(0).text()).toEqual('CLOSE');
        expect(mockClose.mock.calls.length).toBe(1);
    });



    //shows the close button
    //close button is fired
});