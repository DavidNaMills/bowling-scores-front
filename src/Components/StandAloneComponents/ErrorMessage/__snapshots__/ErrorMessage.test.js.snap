// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> test suite snapshot test 1`] = `
<div
  className="errorMessage"
>
  <p
    className="errorMessage__msg"
  >
    I am a long test message
  </p>
</div>
`;
